erpnext/stock/doctype/item/item.py

```diff
@@ -806,6 +807,78 @@ def validate_has_variants(self):
 			if frappe.db.exists("Item", {"variant_of": self.name}):
 				frappe.throw(_("Item has variants."))
 
+	def validate_attributes_in_variants(self):
+		if not self.has_variants:
+			return
+
+		old_doc = self.get_doc_before_save()
+		old_doc_attributes = set([attr.attribute for attr in old_doc.attributes])
+		own_attributes = [attr.attribute for attr in self.attributes]
+
+		# Check if old attributes were removed from the list
+		# Is old_attrs is a subset of new ones
+		# that means we need not check any changes
+		if old_doc_attributes.issubset(set(own_attributes)):
+			return
+
+		from collections import defaultdict
+
+		# get all item variants
+		items = [item["name"] for item in frappe.get_all("Item", {"variant_of": self.name})]
+		items_string = ', '.join(items)
+
+		# get all deleted attributes
+		deleted_attribute = list(old_doc_attributes.difference(set(own_attributes)))
+
+		# fetch all attributes of these items
+		item_attributes = frappe.get_all(
+			"Item Variant Attribute",
+			filters={
+				"parent": ["in", "({})".format(items_string)],
+				"attribute": ["in", "({})".format(', '.join(deleted_attribute))]
```

Comment:
```suggestion
				"parent": ["in", items],
				"attribute": ["in", deleted_attribute]
```
**in** has to be checked with a list or tuple not a string. it returns no results with a string